MdeModulePkg/Universal/Variable/Pei/Variable.c

```diff
@@ -1042,14 +1043,29 @@ PeiGetVariable (
     return EFI_NOT_FOUND;
   }
 
+  // MU_CHANGE
+  DEBUG ((DEBUG_VARIABLE, "Enter: FunctionName(%a) FunctionPointer(%p) VariableName(%s) VendorGuid(%g) &Attributes(0x%p) DataSize(0x%Lx) &Data(%p) \n",
+    __FUNCTION__,
+    PeiGetVariable,
+    VariableName,
+    VariableGuid,
+    Attributes,
+    (UINT64)*DataSize,
+    Data));
+
+  // MU_CHANGE
+  PERF_VARIABLE_BEGIN(VariableGuid, VariableName, PeiGetVariable);

```

Comment:
PEI normally has a fixed number of FPDT entries allowed and is relatively space constrained in pre-memory.

If you haven't already, please try this on a real platform and check the impact on FPDT record count in PEI. First boot can sometimes have more reads than subsequent boots so it'd be good to look at both.